.github/workflows/new-project.yml

```diff
@@ -0,0 +1,75 @@
+name: Create new project and run tests etc.
+
+on:
+  push:
+    branches:
+    - dev_refactor
+  pull_request:
+    branches:
+    - dev_refactor
+
+jobs:
+  new-project:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9.7
+    - uses: conda-incubator/setup-miniconda@v2
+      with:
+        miniconda-version: "latest"
+        channels: conda-forge
+        channel-priority: flexible
+        show-channel-urls: true
+    - name: Create blueprint run env from unpinned reqs
+      run: |
+        conda env create --name blueprint --file requirements/requirements.yml
+    - name: Create new project
+      run: |
+        conda run --name blueprint cookiecutter --no-input .
+    - name: Prepare new project
+      working-directory: ./flying_circus
+      run: |
+        git init
+        git config user.name "Monty Python"
+        git config user.email "monty.python@meteoswiss.ch"
+        git add .
+        git commit -m "initial commit"
+    - name: Create dev env from unpinned reqs
+      working-directory: ./flying_circus
+      run: |
+        conda env create --name flying-circus-dev --file requirements/requirements.yml
+        conda env update --name flying-circus-dev --file requirements/dev-requirements.yml
+    - name: Install package into env
+      working-directory: ./flying_circus
+      run: |
+        conda run --name flying-circus-dev python -m pip install . --no-deps
+    - name: Run pre-commit hooks
+      working-directory: ./flying_circus
+      run: |
+        conda run --name flying-circus-dev pre-commit install-hooks
+        conda run --name flying-circus-dev pre-commit run --all-files
+    - name: Run formatters
+      working-directory: ./flying_circus
+      run: |
+        conda run --name flying-circus-dev black .
+        conda run --name flying-circus-dev isort .
+    - name: Run type checker
+      working-directory: ./flying_circus
+      run: |
+        conda run --name flying-circus-dev mypy
+    - name: Run linters
+      working-directory: ./flying_circus
+      run: |
+        conda run --name flying-circus-dev pydocstyle
+        conda run --name flying-circus-dev flake8
+        conda run --name flying-circus-dev pylint src tests
+    - name: Run unit tests
+      working-directory: ./flying_circus
+      run: |
+        conda run --name flying-circus-dev pytest
```

Comment:
If I understand correctly, this triggers the creation and testing of a new "dummy" project through pre-commit. We already do this in Jenkins upon merges to the main branch. I think we should decide if we do it with Jenkins or pre-commit, but doing it twice seems to be a bit much to me.

Reply:
ah yes sorry. Anyways Jenkins is a mandatory reviewer for pushes to the main. But if it works reasonably well, we could even think about dropping the corresponding action in Jenkins. However I didn't really think this through yet and I'd also be curious on what @cosunae thinks.